python/lsst/ts/ess/common/sensor/campbell_csat3b.py

```diff
@@ -0,0 +1,170 @@
+        b = ((lsb << 1) + msb + ord(char)) % 256
+        if lsb & 0x80:
+            b = b + 1
+        msb = lsb
+        lsb = b
+    # Ensure that the return value is in the range of a C unsigned word.
+    return ((msb << 8) + lsb) % 65536
+
+
+class Csat3bSensor(BaseSensor):
+    """CSAT3B Sensor.
+
+    Perform protocol conversion for Campbell Scientific CSAT3B Anemometer
+    instruments. Serial data is output by the anemometer once per 100 ms with
+    the following format:
+
+        'X,Y,Z,T,D,C,S<CR>'
+
+    where:
+
+        X         x-axis wind speed (m/s). Decimal value.
+        Y         y-axis wind speed (m/s). Decimal value.
+        Z         z-axis wind speed (m/s). Decimal value.
+        T         Sonic temperature (degrees C). Decimal value.
+        D         Diagnostic word. Single digit decimal.
+        C         Record counter. One or two digit value (0-63).
+        S         Signature. Four character hexadecimal value.
+        <CR>      1-character terminator ('\r').
+
+    Parameters
+    ----------
+    log : `logger`
+        The logger for which to create a child logger.
+    """
+
+    def __init__(
+        self,
+        log: logging.Logger,
+    ) -> None:
+        super().__init__(log=log)
+
+        # Override default value.
+        self.terminator = "\r"
+        # Override default value.
+        self.charset = "ISO-8859-1"
+
+    async def extract_telemetry(self, line: str) -> list[float | int]:
+        """Extract the telemetry from a line of Sensor data.
+
+        Parameters
+        ----------
+        line : `str`
+            A line of comma separated telemetry as described in the doc string
+            of this class.
+
+        Returns
+        -------
+        `list`
+            A list of floats and ints containing the telemetry as measured by
+            the sensor. See the description in the doc string of this class.
+        """
+        stripped_line: str = line.strip(self.terminator)
+        line_items = stripped_line.split(self.delimiter)
+        output: list[float | int] = []
+        if len(line_items) == _NUM_VALUES:
+            x = float(line_items[0])
+            y = float(line_items[1])
+            z = float(line_items[2])
+            t = float(line_items[3])
+            d = int(line_items[4])
+            c = int(line_items[5])
+            s = int(line_items[6], 16)
+
+            # Prepare string for signature computation by removing the
+            # signature as reported by the sensor.
+            last_index = line.rfind(self.delimiter)
+            input_line = line[:last_index]
+
+            # Compute signature.
+            signature = compute_signature(input_line, self.delimiter)
+
+            if signature != s:
+                # The computed signature doesn't match the one returned by the
+                # sensor so the telemetry is invalid.
+                output = np.full(_NUM_VALUES, np.nan).tolist()
```

Comment:
I suggest logging this, including the full reply. I hope you will find it to be quite rare.

Reply:
Please consider computing this "nan output" once as a named constant, e.g.:
```
_NANS_OUTPUT = [math.nan] * _NUM_VALUES
```
(or `np.nan` if you prefer, but I don't think you are using numpy for anything else in this module, and math is built in)

Reply:
Done